src/kakarot/eth_rpc.cairo

```diff
@@ -221,3 +224,93 @@ func eth_send_transaction{
 
     return result;
 }
+
+@external
+func eth_send_raw_transaction{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(tx_data_len: felt, tx_data: felt*) -> (
+    return_data_len: felt, return_data: felt*, success: felt, gas_used: felt
+) {
+    let tx = EthTransaction.decode(tx_data_len, tx_data);
+
+    // Whitelisting pre-eip155 or validate chain_id for post eip155
+    let (chain_id) = Kakarot.eth_chain_id();
+    tempvar is_tx_post_eip155 = is_not_zero(tx.chain_id.is_some);
+    if (is_tx_post_eip155 != FALSE) {
+        with_attr error_message("Invalid chain id") {
+            assert tx.chain_id.value = chain_id;
+        }
+        tempvar syscall_ptr = syscall_ptr;
+        tempvar pedersen_ptr = pedersen_ptr;
+        tempvar range_check_ptr = range_check_ptr;
+    } else {
+        tempvar syscall_ptr = syscall_ptr;
+        tempvar pedersen_ptr = pedersen_ptr;
+        tempvar range_check_ptr = range_check_ptr;
+    }
+    let syscall_ptr = cast([ap - 3], felt*);
+    let pedersen_ptr = cast([ap - 2], HashBuiltin*);
+    let range_check_ptr = [ap - 1];
+
+    // Get the caller address
+    let (caller_address) = get_caller_address();
+
+    // Validate nonce
+    let (account_nonce) = IAccount.get_nonce(contract_address=caller_address);
+    with_attr error_message("Invalid nonce") {
+        assert tx.signer_nonce = account_nonce;
+    }
+
+    // Validate gas and value
+    let (kakarot_address) = Ownable_owner.read();
+    let (native_token_address) = Kakarot.get_native_token();
+    let (balance) = IERC20.balanceOf(native_token_address, caller_address);
+
+    with_attr error_message("Gas limit too high") {
+        assert_le_felt(tx.gas_limit, 2 ** 64 - 1);
+    }
+
+    with_attr error_message("Max fee per gas too high") {
+        assert [range_check_ptr] = tx.max_fee_per_gas;
+        let range_check_ptr = range_check_ptr + 1;
+    }
+
+    Kakarot.get_block_gas_limit();
+    tempvar block_gas_limit = [[ap - 1]];
```

Comment:
Solution to avoid error from `is_nn` hint 
If we use 
```
 let (block_gas_limit) = Kakarot.get_block_gas_limit();
 ```
`block_gas_limit` is a `RelocatableValue` leading to throwing in `is_nn`
 ```
  Traceback (most recent call last):
E             File "/Users/yo/Library/Caches/pypoetry/virtualenvs/kakarot-wVIWwPGe-py3.10/lib/python3.10/site-packages/starkware/cairo/common/math_cmp.cairo", line 27, in <module>
E               %{ memory[ap] = 0 if 0 <= ((-ids.a - 1) % PRIME) < range_check_builtin.bound else 1 %}
E           TypeError: bad operand type for unary -: 'RelocatableValue'
 ```

Reply:
I see no reason why it would be so. You can also read the storage var directly it if helps. Surprised by this tbh

Reply:
Yes me neither, this is weird.
Will try to read the storage var directly to see if this fix the problem.

Reply:
```
diff --git a/src/kakarot/eth_rpc.cairo b/src/kakarot/eth_rpc.cairo
index fd69bf5f..882dc32f 100644
--- a/src/kakarot/eth_rpc.cairo
+++ b/src/kakarot/eth_rpc.cairo
@@ -14,7 +14,7 @@ from kakarot.account import Account
 from kakarot.interfaces.interfaces import IAccount, IERC20
 from kakarot.library import Kakarot
 from kakarot.model import model
-from kakarot.storages import Kakarot_native_token_address
+from kakarot.storages import Kakarot_native_token_address, Kakarot_block_gas_limit
 from utils.eth_transaction import EthTransaction
 from utils.maths import unsigned_div_rem
 from utils.utils import Helpers
@@ -275,8 +275,7 @@ func eth_send_raw_transaction{
         let range_check_ptr = range_check_ptr + 1;
     }

-    Kakarot.get_block_gas_limit();
-    tempvar block_gas_limit = [[ap - 1]];
+    let (block_gas_limit) = Kakarot_block_gas_limit.read();
     tempvar tx_gas_fits_in_block = is_nn(block_gas_limit - tx.gas_limit);
     with_attr error_message("Transaction gas_limit > Block gas_limit") {
         assert tx_gas_fits_in_block = TRUE;
 ```
 
 looks like it works this way...

Reply:
Same thing with reading directly the storage variable with 
```
let (block_gas_limit) = Kakarot_block_gas_limit.read();
```
`block_gas_limit` is still a `RelocatableValue`

Reply:
Found the problem: it was my fault for wrongly patching the storage var ...